test/spec/unit/remote.js

```diff
@@ -157,5 +158,104 @@ describe('remote method', () => {
 
             return assert.isRejected(client.getUrl(), /some-error/)
         })
+
+        it('should not change connection retry options if "screenshotOnReject" is boolean', () => {
+            var client = remote({screenshotOnReject: true})
+            var {connectionRetryTimeout, connectionRetryCount} = client.options
+
+            return client.getUrl()
+                .catch(() => {
+                    assert.strictEqual(client.options.connectionRetryTimeout, connectionRetryTimeout)
+                    assert.strictEqual(client.options.connectionRetryCount, connectionRetryCount)
+                })
+        })
+
+        it('should set default connection retry options if "screenshotOnReject" is empty object', () => {
+            var client = remote({screenshotOnReject: {}})
+            var defaultConnectionOpts = {
+                connectionRetryTimeout: client.options.connectionRetryTimeout,
+                connectionRetryCount: client.options.connectionRetryCount
+            }
+
+            return client.getUrl()
+                .catch(() => assert.calledWith(RequestHandler.prototype.updateConnectionOpts, defaultConnectionOpts))
+        })
+
+        it('should set specified connection retry options', () => {
+            var connectionOpts = {
+                connectionRetryTimeout: 30000,
+                connectionRetryCount: 0
+            }
+            var client = remote({screenshotOnReject: connectionOpts})
+
+            return client.getUrl()
+                .catch(() => assert.calledWith(RequestHandler.prototype.updateConnectionOpts, connectionOpts))
+        });
+
+        ['completed successfully', 'failed'].forEach((message) => {
+            it(`should reset connection retry options to default if takeScreenshot is ${message}`, () => {
+                var client = remote({
+                    screenshotOnReject: {
+                        connectionRetryTimeout: 30000,
+                        connectionRetryCount: 0
+                    }
+                })
+
+                var defaultConnectionRetryTimeout = client.options.connectionRetryTimeout
+                var defaultConnectionRetryCount = client.options.connectionRetryCount
+
+                var takeScreenshot = RequestHandler.prototype.create.withArgs('/session/:sessionId/screenshot')
+
+                if (message === 'failed') {
```

Comment:
thats a bad practice to join tests and then add invariants in them. You should split them into separate test cases. There will be some duplication, but thats acceptable for tests and also you can use some helper to decrease duplication